src/evict/evict_lru.c

```diff
@@ -303,79 +303,81 @@ __wt_evict_thread_run(WT_SESSION_IMPL *session, WT_THREAD *thread)
 		__wt_epoch(session, &cache->stuck_ts);
 #endif
 
-	for (;;) {
-		if (!F_ISSET(thread, WT_THREAD_RUN))
-			break;
-		if (F_ISSET(thread, WT_THREAD_ACTIVE)) {
-			if (conn->evict_server_running &&
-			    __wt_spin_trylock(
-			    session, &cache->evict_pass_lock) == 0) {
-				/*
-				 * Cannot use WT_WITH_PASS_LOCK because this is
-				 * a try lock.  Fix when that is supported.  We
-				 * set the flag * on both sessions because we
-				 * may call clear_walk when we are walking with
-				 * the walk session, locked.
-				 */
-				F_SET(session, WT_SESSION_LOCKED_PASS);
-				F_SET(cache->walk_session,
-				    WT_SESSION_LOCKED_PASS);
-				ret = __evict_server(session, &did_work);
-				F_CLR(cache->walk_session,
-				    WT_SESSION_LOCKED_PASS);
-				F_CLR(session, WT_SESSION_LOCKED_PASS);
-				was_intr = cache->pass_intr != 0;
-				__wt_spin_unlock(session,
-				    &cache->evict_pass_lock);
-				WT_ERR(ret);
+	if (conn->evict_server_running &&
```

Comment:
I actually like this structure more, but I think it deserves a comment explaining that the thread group repeatedly calls this function.